src/modules/Ultrastar/ultrastar_converter.py

```diff
@@ -48,32 +54,135 @@ def ultrastar_note_to_midi_note(ultrastar_note: int) -> int:
     return midi_note
 
 
-def get_start_time_from_ultrastar(ultrastar_class: UltrastarTxtValue, pos: int) -> float:
+def get_start_time_from_ultrastar(
+    ultrastar_class: UltrastarTxtValue, pos: int
+) -> float:
     """Calculates the start time from the Ultrastar txt"""
 
-    gap = int(ultrastar_class.gap) / 1000
-    real_bpm = ultrastar_bpm_to_real_bpm(
-        float(ultrastar_class.bpm.replace(",", "."))
-    )
-    start_time = (
-        beat_to_second(int(ultrastar_class.startBeat[pos]), real_bpm) + gap
-    )
+    gap = int(float(ultrastar_class.gap.replace(",", ".")) / 1000)
+    real_bpm = ultrastar_bpm_to_real_bpm(float(ultrastar_class.bpm.replace(",", ".")))
+    start_time = beat_to_second(int(ultrastar_class.startBeat[pos]), real_bpm) + gap
     return start_time
 
 
 def get_end_time_from_ultrastar(ultrastar_class: UltrastarTxtValue, pos: int) -> float:
     """Calculates the end time from the Ultrastar txt"""
 
-    gap = int(ultrastar_class.gap) / 1000
-    real_bpm = ultrastar_bpm_to_real_bpm(
-        float(ultrastar_class.bpm.replace(",", "."))
-    )
+    gap = int(float(ultrastar_class.gap.replace(",", ".")) / 1000)
```

Comment:
This breaks the score calculation

Reply:
good catch.
another contributing factor to broken score calculation was the wrong usage of the constructor here:
https://github.com/rakuri255/UltraSinger/pull/146/commits/76a495653bc4c7ccd21b5a448bc4362de39efa96#diff-f8287eea965b098ba5cef0e0b678bd57ea9d41c3985c3a311c940792de15c57fR987

Calling the constructor like this
```python
new_transcribed_data.append(TranscribedData({"word": midi_segment.word, "start": midi_segment.start, "end": midi_segment.end, "is_hyphen": None, "confidence": 1}))
```

just initializes the confidence field to have the map as it's value and all other fields will have their default value. So all start and end values are 0 from then on.

Reply:
resolved